classes/local/datasource/restore/readme.txt

```diff
@@ -0,0 +1,11 @@
+A short explanation for these classes.
+
+The restore system requires the infrastructure to be set up at the beginning, before anything is known about the data.
+In order to do this, I needed to make a special case of creating all the datasource objects with a generic blank CMS object.
+
+Data source objects are designed to work with a CMS object, or at least a CMS type. Using one with a blank CMS is problrmatic and
```

Comment:
Typo? Not sure what _problrmatic_ means heh

Reply:
It's like problematic, but with an R.